File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        l = len(nums)
        ans = []
        def rec(i, lst):
            if i==l:
                ans.append(lst)
                return
            #not pick
            rec(i+1,lst.copy())
            
            #pick
            lst.append(nums[i])
            rec(i+1,lst.copy())
            
        rec(0,[])
        return ans

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> temp;
        helper(0, nums, ans, temp);
        return ans;
    }

    void helper(int i, vector<int>& nums, vector<vector<int>>& ans, vector<int>& temp ){
        if( i == nums.size() ){
            ans.push_back(temp);
            return ;
        }
        temp.push_back(nums[i]);
        helper(i+1, nums, ans, temp);

        temp.pop_back();
        helper(i+1, nums, ans, temp);
    }
};